sfge/src/main/java/com/salesforce/rules/ThreadableRuleExecutor.java

```diff
@@ -182,21 +179,29 @@ public void run() {
                             new Violation.TimeoutViolation(
                                     "Path evaluation timed out after " + TIMEOUT + " ms",
                                     submission.getPathEntry()));
+                    if (LOGGER.isErrorEnabled()) {
+                        LOGGER.error("Timeout Error executing rule. submission=" + submission, ex);
+                    }
                 } else {
                     // If the thread threw another exception, wrap it in a violation, including the
-                    // last line that executed
+                    // last 3 frames that got executed.
+                    String frameString =
+                            Arrays.stream(ex.getStackTrace())
+                                    .limit(3)
+                                    .map(StackTraceElement::toString)
+                                    .collect(Collectors.joining(";"));
                     final String details =
                             ex.getClass().getSimpleName()
                                     + ": "
                                     + ex.getMessage()
                                     + ": "
-                                    + ex.getStackTrace()[0];
+                                    + frameString;
                     violations.add(
                             new Violation.InternalErrorViolation(
                                     details, submission.getPathEntry()));
-                }
-                if (LOGGER.isErrorEnabled()) {
-                    LOGGER.error("Error executing rule. submission=" + submission, ex);
+                    if (LOGGER.isErrorEnabled()) {
+                        LOGGER.error("Internal Error executing rule. submission=" + submission, ex);
```

Comment:
This should make it easier to find internal errors in customer logs if we need to look.